src/commands/common/get-latest-service-agreement/blockchain-get-latest-service-agreement.js

```diff
@@ -63,39 +63,36 @@ class BlockchainGetLatestServiceAgreement extends Command {
             `Get latest service agreement: Latest token id on chain: ${latestBlockchainTokenId}, latest token id in database: ${latestDbTokenId} on blockchain: ${blockchain}`,
         );
 
-        const missingTokenIds = Array.from(
-            { length: latestBlockchainTokenId - latestDbTokenId },
-            (_, index) => latestDbTokenId + index + 1,
-        );
-        this.logger.debug(
-            `Get latest service agreement: Found ${missingTokenIds.length} on blockchain: ${blockchain}`,
-        );
-        let batchNumber = 0;
-        while (batchNumber * BATCH_SIZE < missingTokenIds.length) {
-            const promises = [];
-            for (
-                let i = batchNumber * BATCH_SIZE;
-                i < missingTokenIds.length && i < (batchNumber + 1) * BATCH_SIZE;
-                i += 1
+        let tokenIdDifference = latestBlockchainTokenId - latestDbTokenId;
+        let getAgreementDataPromise = [];
+        for (
+            let tokenIdToBeFetched = latestDbTokenId + 1;
+            tokenIdToBeFetched <= latestBlockchainTokenId;
+            tokenIdToBeFetched += 1
+        ) {
+            getAgreementDataPromise.push(
+                this.getAgreementDataForToken(tokenIdToBeFetched, blockchain, contract),
+            );
+            if (
+                getAgreementDataPromise.length === tokenIdDifference ||
+                getAgreementDataPromise.length === BATCH_SIZE
             ) {
-                const tokenIdToBeFetched = missingTokenIds[i];
-                promises.push(
-                    this.getAgreementDataForToken(tokenIdToBeFetched, blockchain, contract),
+                // eslint-disable-next-line no-await-in-loop
+                const missingAgreements = await Promise.all(getAgreementDataPromise);
+
+                // eslint-disable-next-line no-await-in-loop
+                await this.repositoryModuleManager.bulkCreateServiceAgreementRecords(
+                    missingAgreements.filter((agreement) => agreement != null),
                 );
+                getAgreementDataPromise = [];
+                tokenIdDifference -= BATCH_SIZE;
             }
-
-            // eslint-disable-next-line no-await-in-loop
-            const missingAgreements = await Promise.all(promises);
-
-            // eslint-disable-next-line no-await-in-loop
-            await this.repositoryModuleManager.bulkCreateServiceAgreementRecords(
-                missingAgreements.filter((agreement) => agreement != null),
-            );
-            batchNumber += 1;
         }
-        if (missingTokenIds.length !== 0) {
+        if (latestBlockchainTokenId - latestDbTokenId !== 0) {
```

Comment:
Maybe add log that nothing had to be fetched, just to see command is running from logs

Reply:
Before this we have a log that command started and log with values found for last token ids, so we will have all info without adding additional logs